plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection/Utils.php

```diff
@@ -120,38 +121,44 @@ public static function parse_frontend_location_context() {
 			$type        = 'order';
 			$source_data = array( 'orderId' => absint( $wp_query->query_vars['order-received'] ) );
 
-		} elseif ( ( is_cart() || is_checkout() ) && isset( WC()->cart ) && is_a( WC()->cart, 'WC_Cart' ) ) {
```

Comment:
@kmanijak I noticed the frontend checkout detection isn't working as expected (try adding the block to the Checkout block) so I've only limited this to the cart block for now. I can fix in a followup PR the checkout and add it back here as well